hs_core/management/commands/modify_resource_id.py

```diff
@@ -0,0 +1,104 @@
+Modify the resource id of an existing resource
+
+"""
+
+from django.core.management.base import BaseCommand, CommandError
+from django.core.exceptions import ObjectDoesNotExist
+from hs_core.models import BaseResource, short_id
+from hs_core.hydroshare.utils import get_resource_by_shortkey
+from hs_core.hydroshare.hs_bagit import create_bag
+from uuid import UUID
+from django.db import transaction, IntegrityError
+from django_irods.icommands import SessionException
+
+
+class Command(BaseCommand):
+    help = "Modify the resource id of an existing resource"
+
+    def add_arguments(self, parser):
+
+        # a list of resource id's, or none to check all resources
+        parser.add_argument('resource_id', type=str, help=('Required. The existing id (short_id) of'
+                                                           ' the resource'))
+        parser.add_argument('new_resource_id', type=str,
+                            help=('Optional. The new id (short_id) for the resource. A random one '
+                                  'is generated if none is provided. Must be a valid string '
+                                  'representation of a uuid hex'))
+
+    def handle(self, *args, **options):
+
+        if not options['resource_id']:
+            raise CommandError('resource_id argument is required')
+        res_id = options['resource_id']
+        try:
+            res = get_resource_by_shortkey(res_id, or_404=False)
+        except ObjectDoesNotExist:
+            raise CommandError("No Resource found for id {}".format(res_id))
+
+        if options['new_resource_id']:
+            try:
+                UUID(options['new_resource_id'])
+                new_res_id = options['new_resource_id']
+            except Exception as e:
+                raise CommandError('new_resource_id {} must be a valid uuid hex string'
+                                   .format(options['new_resource_id']), e)
+
+            try:
+                if BaseResource.objects.get(short_id=new_res_id):
+                    raise CommandError('resource with id {} already exists'.format(new_res_id))
+            except ObjectDoesNotExist:
+                pass
+        else:
+            new_res_id = short_id()
+
+        storage = res.get_irods_storage()
+
+        if storage.exists(res.bag_path):
+            try:
+                storage.delete(res.bag_path)
+                print("{} deleted".format(res.bag_path))
+            except SessionException as ex:
+                print("{} delete failed: {}".format(res.bag_path, ex.stderr))
+                raise EnvironmentError()
+
+        try:
+            with transaction.atomic():
+                print("Deleting existing bag")
+                res.bags.all().delete()
+                res.setAVU("bag_modified", True)
+                res.setAVU('metadata_dirty', 'true')
+
+                print("Updating BaseResource short_id from {} to {}".format(res_id, new_res_id))
+                res.short_id = new_res_id
```

Comment:
Prior to updating the ```short_id``` of the resource, I think we have to hold onto all the logical_file objects of the resource so that we can update the ```resource``` attribute of each of the logical_file objects to the resource with the updated ```short_id```.

Reply:
Ignore my above comment.